shell/server/drivers/ip.js

```diff
@@ -37,7 +37,24 @@ ByteStreamConnection.prototype.write = function (data) {
 // expectSize not implemented
 // ByteStreamConnection.prototype.expectSize = function (size) { }
 
-IpInterfaceImpl = function () { };
+function IpInterfaceImpl (userId, parentToken, grainId) {
+  // TODO(someday): maybe check if userId is admin? This could break things if we don't allow
+  // handing off capabilties from a deleted/deactivated user, so for now we're just allowing all
+  // admins to revoke any ipInterface/ipNetwork caps from the admin settings page.
+  this.userId = userId;
+  this.parentToken = parentToken;
+  this.grainId = grainId;
+}
+
+makeIpInterface = function (userId, parentToken, grainId) {
```

Comment:
I think it's the case that when `parentToken` is provided, `userId` is irrelevant. Moreover, I think in the case where `userId` is provided, the object is used only to handle a subsequent save() call.

Maybe you can simplify by removing the `userId` parameter entirely. When creating a fresh top-level IpInterface token, create the token directly in the table without creating the object at all.